File: 0089-gray-code/0089-gray-code.js
/**
 * @param {number} n
 * @return {number[]}
 */
var grayCode = function(n) {
    const result = [];
    const totalNumbers = 1 << n;
    
    for (let i = 0; i < totalNumbers; i++) {
        result.push(i ^ (i >> 1));
    }
    
    return result;
};

File: 0089-gray-code/0089-gray-code.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> grayCode(int n) {
        vector<int> result;
        int total_numbers = 1 << n;
        result.reserve(total_numbers);
        
        for (int i = 0; i < total_numbers; ++i) {
            result.push_back(i ^ (i >> 1));
        }
        
        return result;
    }
};